aptos-indexer-processors-sdk/sdk/src/common_steps/transaction_stream_step.rs

```diff
@@ -10,28 +11,37 @@ use aptos_indexer_transaction_stream::{
 use aptos_protos::transaction::v1::Transaction;
 use async_trait::async_trait;
 use mockall::mock;
-use std::time::Duration;
+use std::{marker::PhantomData, time::Duration};
 use tokio::sync::Mutex;
 use tracing::{error, info, warn};
 
 // TransactionStreamStep is establishes a gRPC connection with Transaction Stream
 // fetches transactions, and outputs them for processing. It also handles reconnections with retries.
 // This is usually the initial step in a processor.
-pub struct TransactionStreamStep
-where
-    Self: Sized + Send + 'static,
-{
+pub struct TransactionStreamStep<D> {
     transaction_stream_config: TransactionStreamConfig,
     pub transaction_stream: Mutex<TransactionStreamInternal>,
+    phantom: PhantomData<D>,
 }
 
-impl TransactionStreamStep
+impl<D> TransactionStreamStep<D>
 where
     Self: Sized + Send + 'static,
+    D: DbConfig,
```

Comment:
I disagree that a `DbConfig` is required to have transaction stream step

We shouldn't require `save_processor_status`, `get_processor_status`, `save_chain_id`, `save_backfill_processor_status`, and `get_backfill_processor_status` methods to be defined in order to use the step. 

A concrete example would be the event stream processor, where I don't use a DB at all. I just pass in `None` as the stream starting version to just use [live mode](https://github.com/aptos-labs/aptos-core/pull/13393), save the transactions to IMC, stream it, and not even save the processor status to db